tests/wallet/cat_wallet/test_cat_lifecycle.py

```diff
@@ -36,71 +36,72 @@
+) -> int:
+    if limitations_solutions is None:
+        limitations_solutions = [Program.to([])] * len(coins)
+    if extra_deltas is None:
+        extra_deltas = [0] * len(coins)
+
+    spendable_cat_list: List[SpendableCAT] = []
+    for coin, innersol, proof, limitations_solution, extra_delta in zip(
+        coins, inner_solutions, lineage_proofs, limitations_solutions, extra_deltas
     ):
-        if limitations_solutions is None:
-            limitations_solutions = [Program.to([])] * len(coins)
-        if extra_deltas is None:
-            extra_deltas = [0] * len(coins)
-
-        spendable_cat_list: List[SpendableCAT] = []
-        for coin, innersol, proof, limitations_solution, extra_delta in zip(
-            coins, inner_solutions, lineage_proofs, limitations_solutions, extra_deltas
-        ):
-            spendable_cat_list.append(
-                SpendableCAT(
-                    coin,
-                    tail.get_tree_hash(),
-                    acs,
-                    innersol,
-                    limitations_solution=limitations_solution,
-                    lineage_proof=proof,
-                    extra_delta=extra_delta,
-                    limitations_program_reveal=tail if reveal_limitations_program else Program.to([]),
-                )
+        spendable_cat_list.append(
+            SpendableCAT(
+                coin,
+                tail.get_tree_hash(),
+                acs,
+                innersol,
+                limitations_solution=limitations_solution,
+                lineage_proof=proof,
+                extra_delta=extra_delta,
+                limitations_program_reveal=tail if reveal_limitations_program else Program.to([]),
             )
-
-        spend_bundle: SpendBundle = unsigned_spend_bundle_for_spendable_cats(
-            CAT_MOD,
-            spendable_cat_list,
         )
-        agg_sig = AugSchemeMPL.aggregate(signatures)
-        result = await sim_client.push_tx(
-            SpendBundle.aggregate(
-                [
-                    *additional_spends,
-                    spend_bundle,
-                    SpendBundle([], agg_sig),  # "Signing" the spend bundle
-                ]
-            )
+
+    spend_bundle: SpendBundle = unsigned_spend_bundle_for_spendable_cats(
+        CAT_MOD,
+        spendable_cat_list,
+    )
+    agg_sig = AugSchemeMPL.aggregate(signatures)
+    result = await sim_client.push_tx(
+        SpendBundle.aggregate(
+            [
+                *additional_spends,
+                spend_bundle,
+                SpendBundle([], agg_sig),  # "Signing" the spend bundle
+            ]
         )
-        assert result == expected_result
-        self.cost[cost_str] = cost_of_spend_bundle(spend_bundle)
-        await sim.farm_block()
+    )
+    assert result == expected_result
+    cost = cost_of_spend_bundle(spend_bundle)
+    await sim.farm_block()
+    return cost
+
+
+class TestCATLifecycle:
+    cost: Dict[str, int] = {}
```

Comment:
this looks scary. Do tests share state here? does order matter? what happens if you run different tests in different processes?

Reply:
I had to carefully read this to make sure - `cost` is never used to produce test output - it just accumulates the results.

By the way, the very last test that displays the results .. I'm not sure that pytest guarantees the order the tests are run in, so that may not be correct (it doesn't compare, it just prints the stored results)

Reply:
I see. we should probably just make the tests themselves print the result. or assert that it is what we expect it to be

Reply:
Ya, I'll Ping Matt Hauff @quexington